ml-agents-envs/mlagents/envs/base_unity_environment.py

```diff
@@ -1,19 +1,26 @@
 from abc import ABC, abstractmethod
-from typing import Dict
+from typing import Dict, Optional, Any
 
 from mlagents.envs.brain import AllBrainInfo, BrainParameters
 
 
 class BaseUnityEnvironment(ABC):
     @abstractmethod
     def step(
-        self, vector_action=None, memory=None, text_action=None, value=None
+        self,
+        vector_action: Dict,
```

Comment:
Looks like this needs to be `Optional[Dict] = None`

Reply:
I actually think that, unlike the others, an action is mandatory -- that's why I switched it to `Dict`

Reply:
The unit tests disagree :)

Reply:
I thought it was because the test was wrong, but I now remember a recent change means this should actually be the way it was before :-)